crates/ty_python_semantic/resources/mdtest/pep695_type_aliases.md

```diff
@@ -287,6 +287,20 @@ def _(x: C):
     reveal_type(x)  # revealed: () -> C | None
 ```
 
+### Recursive relation in invariant position
+
+```py
+from typing import Any, Callable
+
+type JsonValue = None | JsonDict
+type JsonDict = dict[str, JsonValue]
+type JsonSchemaExtraCallable = Callable[[JsonDict], None] | Callable[[JsonDict, type[Any]], None]
+
+def _update_class_schema(json_schema: dict[str, Any], json_schema_extra: JsonSchemaExtraCallable) -> None:
+    if isinstance(json_schema_extra, dict):
```

Comment:
This test case is a bit weird, because the `JsonSchemaExtraCallable` alias only contains callables. If this test case fails without your fix, it's fine, but wonder if something important got lost.

Reply:
Yes -- this doesn't simplify because `dict` and `Callable` are not disjoint, so it remains an intersection of `Top[dict[Unknown, Unknown]] & JsonSchemaExtraCallable`, and somehow calling the overloaded `dict.update` with that intersection type is necessary to trigger the issue. I will spend some more time tomorrow understanding the precise mechanism more fully so I can write some more thorough and low-level tests.